src/workflows-app/components/SubmitWorkflowModal.test.ts

```diff
@@ -102,284 +97,170 @@ const postRunSetPayload = expect.objectContaining({
-
-    // ** ACT **
-    // user click on Submit button
-    await user.click(modalSubmitButton);
-
-    // ** ASSERT **
-    // assert no requests made
-    expect(createAppV2).not.toHaveBeenCalled();
-    expect(listAppsV2).not.toHaveBeenCalled();
-    expect(postRunSetFunction).not.toHaveBeenCalled();
-  });
-});
-
-describe('SubmitWorkflowModal submitting to workflows app', () => {
-  beforeEach(() => {
-    workflowsAppStore.reset();
-    asMockedFn(getConfig).mockReturnValue({
-      wdsUrlRoot: 'https://lz-abc/wds-abc-c07807929cd1/',
-      cbasUrlRoot: 'https://lz-abc/terra-app-wfa-abc/cbas',
-      cromwellUrlRoot: 'https://lz-abc/terra-app-abc/cromwell',
-    });
-    asMockedFn(isFeaturePreviewEnabled).mockImplementation((id) => id === ENABLE_AZURE_COLLABORATIVE_WORKFLOW_RUNNERS);
-  });
-
-  it('should allow submit to workflows app by creator with running cromwell-runner', async () => {
-    // ** ARRANGE **
-    const user = userEvent.setup();
-    const postRunSetFunction = jest.fn();
-    const createAppV2 = jest.fn();
-    const listAppsV2 = jest.fn(() => Promise.resolve(mockCollaborativeAzureApps));
-    asMockedFn(getTerraUser).mockReturnValue({ email: 'groot@gmail.com' });
-
-    await asMockedFn<() => DeepPartial<AjaxContract>>(Ajax).mockImplementation(() => {
-      return {
-        Apps: {
-          createAppV2,
-          listAppsV2,
-        },
-        Cbas: {
-          runSets: {
-            post: postRunSetFunction,
-          },
-        },
-      };
-    });
-
-    // ** ACT **
-    await act(async () => render(h(SubmitWorkflowModal, submitModalProps)));
-
-    // ** ASSERT **
-    // Launch modal should be displayed
-    const modalSubmitButton = screen.getByLabelText('Launch Submission');
-    expect(modalSubmitButton).not.toHaveAttribute('disabled');
-
-    // ** ACT **
-    // user click on Submit button
-    await user.click(modalSubmitButton);
-
-    // ** ASSERT **
-    expect(createAppV2).not.toHaveBeenCalled();
-    expect(listAppsV2).toHaveBeenCalledTimes(1);
-    expect(postRunSetFunction).toHaveBeenCalledWith('https://lz-abc/terra-app-wfa-abc/cbas', postRunSetPayload);
-  });
-
-  it('should allow submit to workflows app by writer without a cromwell-runner', async () => {
-    // ** ARRANGE **
-    const user = userEvent.setup();
-    const postRunSetFunction = jest.fn();
-    const createAppV2 = jest.fn();
-    const listAppsV2 = jest
-      .fn()
-      .mockImplementationOnce(() => Promise.resolve([]))
-      .mockImplementation(() => Promise.resolve(mockCollaborativeAzureApps));
-    asMockedFn(getTerraUser).mockReturnValue({ email: 'groot@gmail.com' });
-
-    await asMockedFn<() => DeepPartial<AjaxContract>>(Ajax).mockImplementation(() => {
-      return {
-        Apps: {
-          createAppV2,
-          listAppsV2,
-        },
-        Cbas: {
-          runSets: {
-            post: postRunSetFunction,
+describe('SubmitWorkflowModal', () => {
```

Comment:
This is fantastic. Reduced by 100 lines and gained 9 extra test cases. And it's super easy to see what behavioral difference we're expecting in each case. 💯 !